src/tests/Application.vala

```diff
@@ -169,14 +169,14 @@ namespace Terminal.Test.Application {
         });
 
         GLib.Test.add_func ("/application/command-line/execute", () => {
-            string[] execute = { "true", "echo test", "echo -e te\\tst", "false" };
+            string[] execute = { "true", "echo test", "echo -e te\\tst", "false" }; // 4 commands
 
             // valid
             option ("{'execute':<[b'%s']>}".printf (string.joinv ("',b'", execute)), "@a{sv} {}", () => {
                 unowned var window = (MainWindow) application.active_window;
                 assert_nonnull (window);
                 var n_tabs = (int) window.terminals.length ();
-                assert_cmpint (n_tabs, CompareOperator.EQ, 5); // include the guaranted extra tab
+                assert_cmpint (n_tabs, CompareOperator.EQ, 5); // Includes initial default tab added when no tabs restored.
```

Comment:
now that we aren't creating the extra tab, this should be 4.

Reply:
Hmm then it should be failing - will investigate.

Reply:
So what is happening is that the test environment opens a window without restoring tabs *before* processing the option.  So there is already a default tab in the window and the execute option creates 4 more, one for each command, giving a total of 5.

If you execute multiple commands before any window is opened e.g. with a commandline `io.elementary.terminal -e "true" -e "echo Hi"` then no extra default tab appears.

Reply:
Further investigation indicates that *6* tabs are added during this test but the callback occurs before the last one is added. So the same race between the callback and adding a tab that was noted in the `new-tab` test is happening here as well

Reply:
simillary as the new-tab test, there's 1 tab from first launch, and 4 from the commands, the sixty one seems strange.